Add an A/B test for quick responses in the Help Center

sirino opened the pull request:
## Proposed Changes

I am adding an A/B test to disbale the "Quick Response" screen when a user contacts support through the Help Center. Users in the no_quick_response group will not see a quick response before they are able to email or chat with support.

## Testing Instructions


1.  Test together with D120784-code. Checkout this branch and test locally. Edit https://github.com/Automattic/wp-calypso/blob/e7116dc7eef0893a24df7841139fbfe35eab9277/packages/help-center/src/data/use-support-availability.ts#L35: Set the value to 0 so that eligibility and experiment assignment are fetched from the backend every time you access support.
2. Assign yourself to the experiment calypso_helpcenter_quick_response_deflection_rate as a user in the control group.
3. On a site with a paid plan, click on the "?" in the top right of your dashboard to open the help center
4. Enter a question of your choice and you will be shown a list of recommended resources. Click on the 'Still need help' button.
5. Choose "Live chat" from the list of options to contact support. You may need to wait a while or try again if there is no availability.
6. On the contact from, click 'Continue'.
7. Assert that your are shown a "Quick response" or, if none can be retrieved, a message that says so.
8. Assert that you can click 'Continue' and a chat box opens. Let them know that you're testing.
9. Go back to step 5 and choose 'Email'. Repeat steps 6 and 7. Assert that the 'Continue' button is blue but DO NOT submit an email. Doing so will prevent you from being able to test this step again.
10. Assign yourself to the experiment calypso_helpcenter_quick_response_deflection_rate as a user in the no_quick_response group.
11. Repeat steps 3 to 6. Assert that you are not shown a "Quick response" and that the chat box opens right away.
12. Repeat step 9. You may test submitting an email now.


## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

Commit:
add A/B test

yoavf commented:
@sirino I think having the continue button directly opening chat/sending an email is a bit confusing. The flow was designed to show _something_ after filling out the form and before it's submitted. Initially, that was Sibyl results, and now quick responses.

Would it work for the experiment purposes to test quick responses against Sibyl?
(The Sibyl code is still there, we'd just need to set `enableGPTResponse` based on the experiment variation). 

If not, then I think we should at least also change the CTA in getCTALabel for the `no_quick_response` group from "continue" to "Email Us"/"Chat with us" etc.

jessie-ross commented on packages/help-center/src/components/help-center-contact-form.tsx:
```diff
@@ -186,6 +187,15 @@ export const HelpCenterContactForm = () => {
 	const [ debouncedMessage ] = useDebounce( message || '', 500 );
 	const [ debouncedSubject ] = useDebounce( subject || '', 500 );
 
+	const [ , experimentAssignment ] = useExperiment(
```
We need to handle the loading state, otherwise users that have a loading assignment will get control experience, which will suddenly flick-over once the assignment loads.

Commit:
Update CTA to reflect that user will contact us

sirino commented:
@yoavf I updated the CTA on the button to make sure users understand when they will contact us. Do you think it is better now?
Testing against Sibyl is not an option imo since it will not allow us to measure the absolute impact of quick responses.

Commit:
Fix linting

Commit:
Tabs, not whitespace

yoavf committed:
Switch to getting the experiment variation from the backend

useExperiment cannot be used as-is in Help Center, and fetching the
variation from the front end here requires significant development.

Here, we work around that by getting the variation from the backend while
getting the user chat availability.

Since we don't show the screens affected by the experiment until the chat
status is loaded, we also don't need to make any further changes to
handle the loading state

yoavf committed:
fix CTA labeling

yoavf reviewed (approved):
LGTM 😄 We just to make sure we cleanup the extra code passing the variation here and on WP.com when the experiment is over.
